cli/2024-04-azure-workload-identity.md

```diff
@@ -0,0 +1,233 @@
+
+    Enter the Client ID of the Entra ID Application that you have created for Radius.
+
+    > Enter the Client ID:
+    ```
+
+Radius is now configured to use Workload Identity for authentication with Azure.
+
+#### User story 2: As an infrastructure operator, I can configure the Radius Azure Provider to use Workload Identity for authentication manually.
+
+1. Install the `rad` CLI.
+
+1. Install Radius using `rad install kubernetes`
+    ```
+    rad install kubernetes --set global.azureWorkloadIdentity.enabled=true
+    ```
+
+1. Create a Radius environment and add an Azure cloud provider.
+    ```
+    rad group create default
+    rad env create default
+    rad env update default --azure-subscription-id <subscription-id> --azure-resource-group <resource-group> 
+    ```
+
+1. Register the Entra ID Application using `rad credential register azure`.
+    ```
+    rad credential register azure --client-id <client-id>
+    ```
+
+Radius is now configured to use Workload Identity for authentication with Azure.
+
+## User Experience
+
+We will be adding a new option to the `rad init --full` command to allow the user to choose between service principal and Azure Workload Identity.
+
+We will be adding a new option to the Radius Helm chart to allow the user to enable Azure Workload Identity during installation via `rad install kubernetes --set global.azureWorkloadIdentity.enabled=true`.
+
+**Sample Input:**
+
+See above
+
+**Sample Output:**
+
+See above
+
+**Sample Recipe Contract:**
+
+n/a
+
+## Design
+
+### High Level Design
+
+During installation, if the user enables the Azure Workload Identity feature, the Radius control plane (bicep-de, ucp, applications-rp) pods will be annotated with the `azure.workload.identity/use: "true"` label. The AzWI mutating admission webhook will see this label on the pods and will project a signed service account token to each of the pods. The signed service account token and the credential registered by the user will be used by Radius to authenticate with Azure.
+
+Please see https://azure.github.io/azure-workload-identity/docs/concepts.html for more information on how the AzWI mutating admission webhook works.
+
+### Architecture Diagram
+
+![Architecture Diagram](./2024-04-azure-workload-identity/architecture.png)
+
+### Detailed Design
+
+#### Advantages (of each option considered)
+
+n/a
+
+#### Disadvantages (of each option considered)
+
+n/a
+
+#### Proposed Option
+
+n/a
+
+### API design
+
+We should not need to make any functional changes to the Radius API design for this feature as long as we re-use the `ClientID` field for Azure credentials. We should update the documentation for the API and maybe some of the field names to represent that the `ClientID` field can be used for both service principal and Azure Workload Identity. We should also ensure that the `ClientSecret` field is not marked as required anywhere in the API spec since it is not required for Azure Workload Identity.
+
```

Comment:
Can we dig into this more? I'm a big sceptical that we'd have NO api changes.

Does the code inside UCP or ApplicationsRP need to know which type of credential should be used?